src/runtime/network_monitor.cpp

```diff
@@ -0,0 +1,259 @@
+#include "network_monitor.hpp"
+#include "debug.hpp"
+
+#if UNITY_ANDROID
+#include <jni.h>
+#include <android/api-level.h>
+#endif
+
+#if UNITY_OSX
+#include <SystemConfiguration/SystemConfiguration.h>
```

Comment:
You need update cmake configuration for TransmuteCore to link with SystemConfiguration framework.